contrib/Dockerfile

```diff
@@ -1,32 +1,34 @@
 FROM ubuntu:16.04
 
 RUN set -e && \
-  apt-get update && \
-  apt-get install -y \
-    software-properties-common \
-    wget \
-    ssh && \
-  add-apt-repository ppa:deadsnakes/ppa && \
-  apt-get update && \
-  apt-get -y install \
-    python2.7 \
-    python3.4 \
-    python3.5 \
-    python3.6 \
-    python-dev \
-    python2.7-dev \
-    python3.4-dev \
-    python3.5-dev \
-    python3.6-dev \
-    pypy \
-    python-pip
+    apt-get update && \
+    apt-get install -y \
+      software-properties-common \
+      wget \
+      ssh && \
+    add-apt-repository ppa:deadsnakes/ppa && \
+    apt-get update && \
+    apt-get -y install \
+      python2.7 \
+      python3.4 \
+      python3.5 \
+      python3.6 \
+      python-dev \
+      python2.7-dev \
+      python3.4-dev \
+      python3.5-dev \
+      python3.6-dev \
+      pypy \
+      python-pip
 
 RUN set -e && \
-  pip install tox \
-    mock \
-    lockfile \
-    coverage
+    pip install --no-cache-dir tox \
+      mock \
+      lockfile \
+      coverage
 
 COPY . /libcloud
+
 WORKDIR /libcloud
-CMD tox -e py2.7,pypypy,py3.4,py3.5,py3.6,lint
+
+CMD ["tox", "-e", "py2.7", "pypypy", "py3.4", "py3.5", "py3.6", "lint"]
```

Comment:
As per my comment on the PR - once I confirm it's working, I will go ahead and split into two Dockerfiles - Dockerfile.python3 and Dockerfile.python2.

In the future, we can also get rid of python2 one all together since we don't officially support it anymore anyway.

Reply:
This is my fault, I have n’t tested it because of time.

Reply:
yes, Upgrade to python3 is definitely necessary.